src/main/scala/cromwell/binding/package.scala

```diff
@@ -24,17 +31,38 @@ package object binding {
   type CallInputs = Map[String, WdlValue]
   type CallOutputs = Map[LocallyQualifiedName, WdlValue]
   type HostInputs = Map[String, WdlValue]
-
   type ImportResolver = String => WdlSource
 
   /**
-   * Core data identifying a workflow including its unique ID, its namespace, and strongly typed inputs.
+   * Constructs a representation of a particular workflow invocation
    */
-  case class WorkflowDescriptor(id: WorkflowId, namespace: NamespaceWithWorkflow, wdlSource: WdlSource, wdlJson: WdlJson, actualInputs: WorkflowCoercedInputs) {
+  case class WorkflowDescriptor(id: WorkflowId, sourceFiles: WorkflowSourceFiles) {
+    val workflowOptions = Try(sourceFiles.workflowOptionsJson.parseJson) match {
+      case Success(JsObject(options)) =>
+        if (options.values.nonEmpty && !options.values.exists(_.isInstanceOf[JsString])) {
+          throw new Throwable(s"Workflow ${id.toString} options JSON is not a String -> String map: ${sourceFiles.workflowOptionsJson}")
+        }
+        options map { case (k, v) => k -> v.asInstanceOf[JsString].value }
+      case _ => throw new Throwable(s"Workflow ${id.toString} contains bad workflow options JSON: ${sourceFiles.inputsJson}")
+    }
+
+    val backendType = Backend.from(workflowOptions.getOrElse("default_backend", ConfigFactory.load.getConfig("backend").getString("backend")))
+    val namespace = NamespaceWithWorkflow.load(sourceFiles.wdlSource, backendType.backendType)
     val name = namespace.workflow.name
     val shortId = id.toString.split("-")(0)
+
+    val rawInputs = Try(sourceFiles.inputsJson.parseJson) match {
+      case Success(JsObject(inputs)) => inputs
+      case _ => throw new Throwable(s"Workflow ${id.toString} contains bad inputs JSON: ${sourceFiles.inputsJson}")
+    }
+
+    val coercedInputs = namespace.coerceRawInputs(rawInputs).get
```

Comment:
No, because if I have to debug someone's None.get I'd like to know if they thought that there was a good reason behind doing so or if they were just being lazy. The way in which I'd proceed is different. 

I suppose it's not a big deal, we have unsafe gets, heads and throws all over the place, what's a few more.